backend/functions/src/matches.ts

```diff
@@ -7,6 +7,7 @@ export const detectMatchesCreateSession = functions
   .database.ref('/queues/{difficulty}')
   .onWrite(async (change, context) => {
     const lvl = context.params.difficulty;
-    await matchingCore.processQueue(lvl);
+    const queueState = change.after.val();
```

Comment:
```suggestion
    const queueState = change.after.val() as string[];
```

We know that the `queueState` is always an array of user Ids so we tell typescript that. For `onWrite`, we know that `queueState` will always have an element inside so we do not have to account for `undefined`.